include/aspect/velocity_boundary_conditions/ascii_data.h

```diff
@@ -0,0 +1,356 @@
+           * calculate the interpolation weight of a position
+           *
+           * @param point_weight Weight for the linear interpolation
+           */
+          void
+          calculate_point_weight(double *point_weight,
+                                 const Point<dim> &position,
+                                 const dealii::TableIndices<dim-1> spatial_index) const;
+
+          /**
+          * Convert a tensor of rank 1 and dimension in to rank 1 and
+          * dimension out. If $out < in$ the last elements will be discarded,
+          * if $out > in$ zeroes will be appended to fill the tensor.
+          */
+          template <int in, int out>
+          Tensor<1,out> convert_tensor (const Tensor<1,in> &old_tensor) const;
+
+
+          /**
+           * This function adds a certain data point to the interpolated
+           * surface velocity at this evaluation point.
+           */
+          Tensor<1,dim>
+          add_interpolation_point(const dealii::TableIndices<dim-1> spatial_index,
+                                  const double point_weight[dim-1],
+                                  const double time_weight) const;
+
+
+      };
+    }
+
+
+    /**
+     * A class that implements prescribed velocity boundary conditions
+     * determined from a AsciiData input file.
+     *
+     * @ingroup VelocityBoundaryConditionsModels
+     */
+    template <int dim>
+    class AsciiData : public Interface<dim>, public SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Empty Constructor.
+         */
+        AsciiData ();
+
+        /**
+         * Initialization function. This function is called once at the
+         * beginning of the program. Checks preconditions.
+         */
+        virtual
+        void
+        initialize ();
+
+        /**
+         * A function that is called at the beginning of each time step. For
+         * the current plugin, this function loads the next velocity files if
+         * necessary and outputs a warning if the end of the set of velocity
+         * files is reached.
+         */
+        virtual
+        void
+        update ();
+
+        /**
+         * Return the boundary velocity as a function of position. For the
+         * current class, this function returns value from the text files.
+         */
+        Tensor<1,dim>
+        boundary_velocity (const Point<dim> &position) const;
+
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        void
+        parse_parameters (ParameterHandler &prm);
+
+      private:
+        /**
+         * A variable that stores the current time of the simulation, but
+         * relative to the velocity_file_start_time.
+         */
+        double time_relative_to_vel_file_start_time;
```

Comment:
This is dangerous: you are duplicating information. I think it would be simpler to not have this variable and either compute its value from the current time wherever necessary, or to have a function that takes the current time and returns whatever you need.